backend-api/src/functions/asyncActiveSession/asyncActiveSessionHandler.ts

```diff
@@ -23,6 +32,28 @@ export async function lambdaHandlerConstructor(
     return unauthorizedResponse;
   }
 
+  const jwe = authorizationHeaderResult.value;
+
+  const kmsAdapter = dependencies.kmsAdapter(config.KID_ARN);
+  const tokenService = dependencies.tokenService(kmsAdapter);
+  const getSubFromTokenResult = await tokenService.getSubFromToken(
+    config.STS_JWKS_ENDPOINT,
+    jwe,
+  );
+  if (getSubFromTokenResult.isError) {
+    if (getSubFromTokenResult.value.errorCategory === "CLIENT_ERROR") {
+      logger.log("FAILED_TO_GET_SUB_FROM_SERVICE_TOKEN", {
+        errorMessage: getSubFromTokenResult.value.errorMessage,
+      });
+      return badRequestResponse;
+    }
+
+    logger.log("INTERNAL_SERVER_ERROR", {
```

Comment:
I think we need to chat about this log message name ~ this is still failing to get a sub from the service token. I'm wondering whether we should include the errorCategory in the log somehow

Reply:
Follow-up PR